clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/input/avro/GenericAvroRecordReader.java

```diff
@@ -0,0 +1,82 @@
+package com.linkedin.venice.hadoop.input.avro;
+
+import com.linkedin.venice.exceptions.VeniceException;
+import com.linkedin.venice.hadoop.exceptions.VeniceSchemaFieldNotFoundException;
+import com.linkedin.venice.utils.Utils;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.NoSuchElementException;
+import org.apache.avro.Schema;
+import org.apache.avro.file.DataFileStream;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.IndexedRecord;
+import org.apache.avro.mapred.AvroWrapper;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+/**
+ * This class reads the value field given the key field from an avro file based on the file's schema.
+ * Returns the fields as {@link Object}
+ * Currently reads only a single record containing one or multiple fields
+ */
+public class GenericAvroRecordReader {
+  private static final Logger LOGGER = LogManager.getLogger(GenericAvroRecordReader.class);
+  private final InputStream inputStream;
+  private final DataFileStream avroDataFileStream;
+  private final Schema fileSchema;
+  private final Object avroObject;
+
+  public GenericAvroRecordReader(FileSystem fs, String file) {
+    if (fs != null && file != null) {
+      try {
+        inputStream = fs.open(new Path(file));
+        avroDataFileStream = new DataFileStream(inputStream, new GenericDatumReader());
```

Comment:
To continue on the `SpecificRecord` point, there seems to be a `SpecificDatumReader` class that could be useful

Example `readFromAvro` function here: https://www.javatips.net/api/hadoop-book-master/src/main/java/com/manning/hip/ch3/avro/AvroStockFileRead.java